torchsde/_core/sdeint.py

```diff
@@ -92,22 +96,29 @@ def sdeint(sde: base_sde.BaseSDE,
         atol=atol,
         dt_min=dt_min,
         options=options,
+        logqp=logqp
     )
 
 
-def check_contract(sde, y0, ts, bm, method, names):
+def check_contract(sde, y0, ts, bm, method, names, logqp):
     if names is None:
         names_to_change = {}
     else:
-        names_to_change = {key: names[key] for key in ("drift", "diffusion") if key in names}
+        names_to_change = {key: names[key] for key in ("drift", "diffusion", "prior_drift") if key in names}
     if len(names_to_change) > 0:
         sde = base_sde.RenameMethodsSDE(sde, **names_to_change)
 
-    required_funcs = ("f", "g")
+    required_funcs = ("f", "g", "h") if logqp else ("f", "g")
     missing_funcs = [func for func in required_funcs if not hasattr(sde, func)]
     if len(missing_funcs) > 0:
         raise ValueError(f"sde is required to have the methods {required_funcs}. Missing functions: {missing_funcs}")
 
+    # --- Backwards compatibility: v0.1.1. ---
+    if logqp:
+        sde = base_sde.SDELogqp(sde)
+        y0 = torch.cat((y0, y0.new_zeros(size=(y0.size(0), 1))), dim=1)
+    # ----------------------------------------
+
```

Comment:
This new block should be moved below the noise type and sde type checks, as `SDELogqp` relies on noise type existing.

Reply:
Good point! Done.